apps/backend/src/lib/controller/chip/managed/leaderboard/index.ts

```diff
@@ -63,127 +82,140 @@ ManagedChipClient.prototype.UpdateLeaderboardEntry = async function (
 
 ManagedChipClient.prototype.GetUserLeaderboardPosition = async function (
   username: string,
-  chipIssuer: ChipIssuer
+  chipIssuer: ChipIssuer,
+  entryType: LeaderboardEntryType
 ): Promise<number | null> {
-  try {
-    /*
-    1. Get list of entries for chipIssuer, ordered on descending tap count
-    subquery1 = SELECT * FROM "LeaderboardEntry" WHERE "chipIssuer"='TESTING' ORDER BY "tapCount" DESC
-
-    2. Select username and calculated row number from entries
-    subquery2 = SELECT username, row_number() OVER() as "leaderboardPosition" FROM (subquery1) AS sorted_entries
-
-    3. Select row number based on username
-    query = SELECT row_number FROM (subquery2) AS filtered_entries WHERE "username"='pass2';
-
-    Working query:
-    SELECT "leaderboardPosition" FROM (SELECT username, row_number() OVER() as "leaderboardPosition" FROM (SELECT * FROM "LeaderboardEntry" WHERE "chipIssuer"='TESTING' ORDER BY "tapCount" DESC) AS sorted_entries) AS rowed_sorted_entries WHERE "username"='pass2';
-    */
-
-    const positions: Array<{ leaderboardPosition: number } | null> = await this
-      .prismaClient
-      .$queryRaw`SELECT "leaderboardPosition" FROM (SELECT username, row_number() OVER() as "leaderboardPosition" FROM (SELECT * FROM "LeaderboardEntry" WHERE "chipIssuer"=${chipIssuer} ORDER BY "tapCount" DESC) AS sorted_entries) AS rowed_sorted_entries WHERE "username"=${username}`;
-
-    let length = -1;
-    if (positions && positions.length == 1 && positions[0] !== null) {
-      // Position should always be length 1. 0 indicates the user doesn't exist, and 2 indicates multiple users with same name.
-      return Number(positions[0].leaderboardPosition);
-    } else if (positions && positions.length == 0) {
-      return -1;
-    } else if (positions) {
-      // I think this is overly defensive and the return value will _never_ be null, but doing it just in case -- -1 would indicate null
-      length = positions.length;
-    }
-
-    throw new Error(
-      `Failed to get user leaderboard position, return value of length ${length}`
-    );
-  } catch (error) {
-    console.error("Failed to get leaderboard position:", errorToString(error));
-    throw new Error("Failed to get leaderboard position");
+  const leaderboardEntries = await this.GetTopLeaderboardEntries(
+    chipIssuer,
+    entryType,
+    undefined
+  );
+
+  if (!leaderboardEntries) {
+    return null;
   }
+
+  return (
+    leaderboardEntries.findIndex((entry) => entry.username === username) + 1
```

Comment:
This is great, excellent improvement. Will keep findIndex in the mind for the future.